src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -0,0 +1,294 @@
+
+import java.util.*;
+
+import static com.minecolonies.api.util.constant.Constants.HALF_BLOCK;
+import static com.minecolonies.api.util.constant.Constants.SLIGHTLY_UP;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_CITIZENS;
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_MAX_CITIZENS;
+
+public class CitizenManager implements ICitizenManager
+{
+    /**
+     * List of citizens.
+     */
+    @NotNull
+    private final Map<Integer, CitizenData> citizens = new HashMap<>();
+
+    /**
+     * Variables to determine if citizens have to be updated on the client side.
+     */
+    private boolean isCitizensDirty  = false;
+
+    /**
+     * The highest citizen id.
+     */
+    private int topCitizenId = 0;
+
+    /**
+     * Max citizens without housing.
+     */
+    private int maxCitizens = Configurations.gameplay.maxCitizens;
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound, @NotNull final Colony colony)
+    {
+        maxCitizens = compound.getInteger(TAG_MAX_CITIZENS);
+
+        //  Citizens before Buildings, because Buildings track the Citizens
+        final NBTTagList citizenTagList = compound.getTagList(TAG_CITIZENS, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < citizenTagList.tagCount(); ++i)
+        {
+            final NBTTagCompound citizenCompound = citizenTagList.getCompoundTagAt(i);
+            final CitizenData data = CitizenData.createFromNBT(citizenCompound, colony);
+            citizens.put(data.getId(), data);
+            topCitizenId = Math.max(topCitizenId, data.getId());
+        }
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        compound.setInteger(TAG_MAX_CITIZENS, maxCitizens);
+
+        //  Citizens
+        @NotNull final NBTTagList citizenTagList = new NBTTagList();
+        for (@NotNull final CitizenData citizen : citizens.values())
+        {
+            @NotNull final NBTTagCompound citizenCompound = new NBTTagCompound();
+            citizen.writeToNBT(citizenCompound);
+            citizenTagList.appendTag(citizenCompound);
+        }
+        compound.setTag(TAG_CITIZENS, citizenTagList);
+    }
+
+    @Override
+    public void sendPackets(@NotNull final Set<EntityPlayerMP> oldSubscribers,
+            final boolean hasNewSubscribers,
+            @NotNull final Set<EntityPlayerMP> subscribers,
+            @NotNull final Colony colony)
+    {
+        if (isCitizensDirty || hasNewSubscribers)
+        {
+            for (@NotNull final CitizenData citizen : citizens.values())
+            {
+                if (citizen.isDirty() || hasNewSubscribers)
+                {
+                    subscribers.stream()
+                            .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
+                            .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewCitizenViewMessage(colony, citizen), player));
```

Comment:
WHAT?!? If you do this in each Manager, you are overloading the bus. Please create update packets for each of the managers.

Reply:
Yeah as we talked in Slack, unfortunately, we're doing it this way, and we should definitely make a future PR to fix that. But for now it would make this PR a mess if we'd update both things at the same time

Reply:
Okiedokie